### PR TITLE
[Merged by Bors] - chore: remove some unused variables

### PR DESCRIPTION
**Pull request by adomani**
Found by the unused variable command linter (in progress).

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/GroupWithZero/Defs.lean
Comment (Ruben-VandeVelde) [inline Mathlib/Algebra/GroupWithZero/Defs.lean:172]: Maybe
```suggestion
@[simp] lemma mul_div_cancel_left₀ (b : M₀) {a : M₀} (ha : a ≠ 0) : a * b / a = b := by
```
Though on the other hand, maybe we just shouldn't make these changes here

Comment (adomani) [reply Mathlib/Algebra/GroupWithZero/Defs.lean:172]: I'd prefer to leave these changes to a separate PR, since I would not want to mix variable removals with variable order changes (at least if I notice that I am doing a reorder!).

File: Mathlib/Algebra/GroupWithZero/Defs.lean
Comment (Ruben-VandeVelde) [inline Mathlib/Algebra/GroupWithZero/Defs.lean:164]: If you want to stick to the existing order:
```suggestion
@[simp] lemma mul_div_cancel_right₀ {b : M₀} (a : M₀) (hb : b ≠ 0) : a * b / b = a :=
```

Comment (adomani) [reply Mathlib/Algebra/GroupWithZero/Defs.lean:164]: You are right: I get confused so easily with `variable`!  My reasoning for not wanting to change the order was mostly since I expected potential breaking further downstream.  Here nothing broke...  😊 

I'll leave it up to you, here and in the other place: choose the order that you prefer and it seems like it won't affect much.

File: Mathlib/Algebra/GroupWithZero/Defs.lean
Comment (adomani) [reply Mathlib/Algebra/GroupWithZero/Defs.lean:172]: Let's see if this change works!

File: Mathlib/Algebra/GroupWithZero/Defs.lean
Comment (adomani) [reply Mathlib/Algebra/GroupWithZero/Defs.lean:164]: Let's leave this in alphabetical order, and change the other as well.

**Review by Ruben-VandeVelde — APPROVED** [review]
maintainer merge

**Comment by sgouezel** [comment]
bors r+
Thanks!

**Pull request closed without merging**
